src/db.rs

```diff
@@ -289,6 +321,108 @@ impl DbInner {
 		}
 	}
 
+	fn get_root(&self, col: ColId, key: &[u8]) -> Result<Option<(Vec<u8>, Children)>> {
+		if !self.options.columns[col as usize].multitree {
+			return Err(Error::InvalidConfiguration("Not a multitree column.".to_string()))
+		}
+		if !self.options.columns[col as usize].append_only {
+			return Err(Error::InvalidConfiguration(
+				"get_root can only be called on a column with append_only option.".to_string(),
+			))
+		}
+		let value = self.get(col, key, false)?;
+		if let Some(data) = value {
+			return Ok(Some(unpack_node_data(data)?))
+		}
+		Ok(None)
+	}
+
+	fn get_node(
+		&self,
+		col: ColId,
+		node_address: NodeAddress,
+		external_call: bool,
+	) -> Result<Option<(Vec<u8>, Children)>> {
+		if !self.options.columns[col as usize].multitree {
+			return Err(Error::InvalidConfiguration("Not a multitree column.".to_string()))
+		}
+		if !self.options.columns[col as usize].append_only && external_call {
+			return Err(Error::InvalidConfiguration(
+				"get_node can only be called on a column with append_only option.".to_string(),
```

Comment:
There won’t be any guarantee that the NodeAddress is still valid though, as it might have been removed. The Db won’t even be able to warn if it has happened.
Is there some external guarantee that this won’t happen?

Reply:
Yes, in substrate there's a higher level state pinning mechanism. It prevents the tree from being deleted while there are active readers.

Reply:
Added the option.